### PR TITLE
Add acceptance test for deleting share

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Added webui Acceptance test for checking the delete share button works.

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 1]
```diff
@@ -131,3 +131,31 @@ Feature: Sharing files and folders with internal groups
...
+    When the user deletes share with group "grp1" for the current file
+    Then "grp1" should not be listed in the shared with list
+#    And file "lorem.txt" should not be listed in shared-with-others page on the webUI
+#    And as "user2" file "lorem (2).txt" should not exist
+
+  @yetToImplement
+  Scenario: user shares the file/folder with multiple internal users and delete the share with one user user
+    Given group "grp2" has been created
+    And user "user3" has been added to group "grp2"
+    Given user "user1" has logged in using the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]:
```suggestion
    And user "user1" has logged in using the webUI
```

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 2]
```diff
@@ -83,10 +83,41 @@ module.exports = {
...
+     *
+     * @returns {Promise.<string[]>} Array of autocomplete webElementIds
+     */
+    deleteShareWithUserGroup: function (item) {
+      const util = require('util')
+      const deleteSelector = util.format(this.elements.deleteShareButton.selector, item)
+      return this
+        .useXpath()
+        .waitForElementVisible(deleteSelector)
+        .api.element('xpath', deleteSelector, result => {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: why not using the callback of `waitForElementVisible` itself?

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 3]
```diff
@@ -39,13 +39,12 @@ const userSharesFileOrFolderWithGroup = function (file, sharee) {
   return userSharesFileOrFolderWithUserOrGroup(file, sharee, true)
 }
 
-Given('user {string} has shared file/folder {string} with user {string}', function (sharer, elementToShare, receiver) {
+const shareFileFolder = function (elementToShare, sharer, receiver, shareType = 0, permissions = 31) {
   const params = new URLSearchParams()
-  params.append('shareType', '0') // 0 = user, 1 = group, 3 = public (link), 6 = federated (cloud share).
+  params.append('shareType', shareType) // 0 = user, 1 = group, 3 = public (link), 6 = federated (cloud share).
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/sharingContext.js]: can we use strings here? the numbers are confusing and hard to remember, same for permissions

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 4]
```diff
@@ -131,3 +131,31 @@ Feature: Sharing files and folders with internal groups
...
     And the autocomplete list should not be displayed on the webUI
+
+  @yetToImplement
+  Scenario: user shares the file/folder with a group and delete the share with user
+    Given user "user1" has logged in using the webUI
+    And user "user1" has shared file "lorem.txt" with group "grp1"
+    When the user opens the share dialog for file "lorem.txt"
+    Then "grp1" should be listed in the shared with list
+    When the user deletes share with group "grp1" for the current file
+    Then "grp1" should not be listed in the shared with list
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: best would be to double-check with the API if the share is also really gone in the backend

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js (hunk 2 above)
Comment (dpakach) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: Using the callback, I dont get the webElementId of the delete button so I couldnot do that or am I doing something wrong

Comment (individual-it) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: why do you need the `webElementId` ? you can use the `this.click()` function with the `deleteSelector`

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3573/6

<p>





</p></details>

File: tests/acceptance/stepDefinitions/sharingContext.js (hunk 3 above)
Comment (dpakach) [reply tests/acceptance/stepDefinitions/sharingContext.js]: I think It would be better to do that in new PR.

**Comment by dpakach** [comment]
@individual-it Please review again.

**Comment by individual-it** [comment]
LGTM
@LukasHirt @DeepDiver1975 @felixheidecke please have a look, one id was added and also aria-labels

File: apps/files/src/components/FileSharingSidebar.vue [hunk 5]
```diff
@@ -52,8 +52,8 @@
                 </span>
                 </template>
                 <template slot="actions">
-                  <oc-icon name="delete" @click="onDelete(c)" />
-                  <oc-icon v-if="editing != c" name="edit" @click="onEdit(c)" />
+                  <oc-icon aria-label="Delete share" name="delete" @click="onDelete(c)" />
```
Comment (LukasHirt) [inline apps/files/src/components/FileSharingSidebar.vue]: Needs to be translated.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3625/13

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3625/14

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3626/14

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3626/13

<p>







</p></details>

**Pull request merged**